### DISCUSSION TITLE
Overlay softwares sometime causes unintended acceleration to osu! High Precision Cursor.

### DISCUSSION THREAD

**Discussion by EmiyaSyahriel** [ppy/osu#14973] [Q&A] [answered]
I am not sure if it was the same issue as #14186 that was also occured in Windows. But this might be a different issue as I have the reproduction steps.

**Issue Description :**
High Precison Cursor sometime will accelerate too much when is reaching an overlay that receives cursor event. So far only tested with Rainmeter, But I think it can also caused by other overlay app.

**How to reproduce the issue :**
- Have an overlay that accepts cursor hover event and is always on top
- Run osu!lazer with that overlay obscuring part of osu!'s window (still reproducible even in Exclusive Fullscreen).
- Do some cursor movement (a gameplay or just random) and then move the cursor at a certain speed to that overlay cursor accept area.

**Expected Behaviour :**
The cursor keep moving at the speed it currently moving.

**Current Behaviour :**
The cursor is moving faster than it should when entering that overlay area. only at cursor enter event, moving after that event remains normal. Seems like it was the cursor is teleported to the edge, but sometimes it just moved halfway between the event enter location to the screen edge meaning that it was probably caused by an unintended acceleration.

**Temporary Fixes :**
- Disable that overlay event receiver.
- Move osu! window somewhere not obstructed by the overlay.

**osu!lazer version:**
2021.1005.0 (still reproducible in 2021.1006.0)

**Repro Video :**
- Video 1 (with overlay) : https://drive.google.com/file/d/1n-uSsAHVK6hljGae4boAIRfIw966s6N4/view?usp=sharing
  This issue is triggered and captured at 2:30 and 2:44
- Video 2 (without overlay) : https://drive.google.com/file/d/1ewezSyBJjN8HOUYucnXRPdkB0pqWrzzy/view?usp=sharing
  The same repro at 1:34 and 1:50 didn't trigger the issue.

**Logs :**
[logs.zip](https://github.com/ppy/osu/files/7289727/logs.zip) *(Taken from 2021.1006.0, reproduced at different time location than the video)*

**Comment by peppy** [comment, marked as answer]
Highly doubt this is something that will be fixed. I'd suggest not having other windows overlapping osu!

**Reply by EmiyaSyahriel** [reply]
Well, It's trivial enough to just move that overlay somewhere, disabling it or just playing osu! fullscreen at other monitor in case there is another monitor available.
